### PR TITLE
RELATED: RAIL-2392 Validate localIds in execution definitions

### PR DESCRIPTION
**Pull request by no23reason**
Currently just validates that the localIds are unique among measures and attributes.

This is to improve the error message (before, a generic 400 would be
thrown) and to not try to execute definitions that are certain to fail.

This helped us find a few test fixtures that were invalid, so they were fixed.

JIRA: RAIL-2392

---

Supported PR commands:

| Command                | Description            |
| ---------------------- | ---------------------- |
| `ok to test`           | Re-run standard checks |
| `extended test`        | BackstopJS tests       |
| `extended check sonar` | SonarQube tests        |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4264/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5877/ : FAILURE in 14m 11s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4270/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5880/ : SUCCESS in 15m 14s

File: libs/sdk-model/src/execution/executionDefinition/factory.ts [hunk 1]
```diff
@@ -70,6 +71,8 @@ export function newDefForItems(
         measures: items.filter(isMeasure),
     };
 
+    validateExecDefinition(def);
```
Comment (lupko) [inline libs/sdk-model/src/execution/executionDefinition/factory.ts:74]: i suggest to rename this to match conventions in this area.. call it something like `defValidate` (e.g. we have `defFingerprint` already)

Comment (no23reason) [reply libs/sdk-model/src/execution/executionDefinition/factory.ts]: Good point, updated

**Review by lupko — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result
 - No issue found. Congratulations!

More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1488

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1226/ : SUCCESS in 1m 23s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1502/ : SUCCESS in 20s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/257/ : SUCCESS in 8m 13s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5885/ : SUCCESS in 13m 56s

**Pull request merged**